lottie/src/main/java/com/airbnb/lottie/parser/GradientColorParser.java

```diff
@@ -223,4 +221,60 @@ private int getColorInBetweenOpacityStops(float position, int color, float[] opa
     }
     throw new IllegalArgumentException("Unreachable code.");
   }
+
+  /**
+   * Takes two sorted float arrays and merges their elements while removing duplicates.
+   */
+  protected static float[] mergeUniqueElements(float[] arrayA, float[] arrayB) {
+    if (arrayA.length == 0) {
+      return arrayB;
+    } else if (arrayB.length == 0) {
+      return arrayA;
+    }
+
+    int aIndex = 0;
+    int bIndex = 0;
+    int mergedIndex = 0;
+    int numDuplicates = 0;
+    // This will be the merged list but may be longer than what is needed if there are duplicates.
+    // If there are, the 0 elements at the end need to be truncated.
+    float[] mergedNotTruncated = new float[arrayA.length + arrayB.length];
+    for (int i = 0; i < mergedNotTruncated.length; i++) {
+      final float a;
+      if (aIndex < arrayA.length) {
+        a = arrayA[aIndex];
+      } else {
+        a = Float.NaN;
+      }
+
+      final float b;
+      if (bIndex < arrayB.length) {
+        b = arrayB[bIndex];
+      } else {
+        b = Float.NaN;
+      }
+
+      if (Float.isNaN(b) || a < b) {
+        mergedNotTruncated[mergedIndex++] = a;
```

Comment:
isn't `mergeIndex` can be replaced by `i`?

Reply:
Yup!